src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -982,6 +983,83 @@ class TcgStatics {
+                    def energyCard
+                    def energyEquivalent = []
+                    def typeImages = []
+                    rainbowEnergyCount.times {
+                      energyEquivalent.add([R, D, F, G, W, L, M, P, Y])
+                      typeImages.add(RAINBOW)
+                    }
+
+                    energyCard = specialEnergy(new CustomCardInfo(thisCard.staticInfo).setCardTypes(ENERGY, SPECIAL_ENERGY), energyEquivalent) {
+                      typeImagesOverride = typeImages
+                      onPlay {}
+                      onRemoveFromPlay {
+                        bg.em().run(new ChangeImplementation(thisCard, energyCard))
+                      }
+                    }
+                    energyCard.player = thisCard.player
+
+                    def pcs = validPokemon.select("Attach to?")
+
+                    // Select an energy before attachment so that thisCard doesn't show up as an option
+                    def returningEnergy = pcs.cards.filterByType(ENERGY).select(count: 1)
+
+                    bg.em().run(new ChangeImplementation(energyCard, thisCard))
+                    def cannotPlayEnergy = bg.em().run(new PlayEnergy(energyCard, pcs))
+                    if (cannotPlayEnergy) {
+                      bg.em().run(new ChangeImplementation(thisCard, energyCard))
+                    } else {
+                      returningEnergy.moveTo(thisCard.player.pbg.hand)
+
+                      bg.em().storeObject("Holon_Pokemon_Energy", bg.turnCount)
+                      bc "$energyCard is now a Special Energy Card"
+                    }
+                  } else {
+                    wcu "You have no energy attached to your Pokémon"
+                  }
+                } else {
+                  wcu "You have already played an Energy"
+                }
+                prevent()
+              }
+            } //If the user chooses Pokémon, play the card normally
+          }
+          before PLAY_ENERGY, {
+            def isFromAbility = bg.em().currentEffectStack.find{it instanceof TcgBuilders.AbilityBuilder}
+            def isEnergyAlreadyPlayed = bg.em().currentEffectStack.find{it instanceof CantPlayEnergy}
+            def holon_played = bg.em().retrieveObject("Holon_Pokemon_Energy")
+            if (holon_played == bg.turnCount) {
+              wcu "Cannot play any more energy this turn."
+              prevent()
+            }
+
+            // allow attachments that aren't from ability
+            // disallow this action if already attached for the turn
+            else if (!isFromAbility && isEnergyAlreadyPlayed && ef.card.name == thisCard.name) {
+              wcu "Cannot play any more energy this turn."
+              prevent()
+            }
+          }
+          after ATTACH_ENERGY, {
+            if (ef.card.name == thisCard.name) {
+              bg.em().storeObject("Holon_Pokemon_Energy", bg.turnCount)
+            }
+          }
```

Comment:
These shouldn't be required at all when using PlayEnergy as well.

Reply:
This has been entirely removed, you're correct that this is unneeded when using PlayEnergy